packages/clay-modal/src/ClayModal.tsx

```diff
@@ -0,0 +1,206 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import Button from '@clayui/button';
+import classNames from 'classnames';
+import Icon from '@clayui/icon';
+import React, {FunctionComponent, useEffect, useRef, useState} from 'react';
+import {createPortal} from 'react-dom';
+import {Footer, FooterContainer, FooterContainerProps} from './Footer';
+import {useModal} from './Hook';
+
+type Size = 'full-screen' | 'lg' | 'sm';
+
+type Status = 'danger' | 'info' | 'success' | 'warning';
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Callback called to close the modal.
+	 */
+	onClose: () => void;
+
+	/**
+	 * Renders the footer with the FooterContainer component utilities.
+	 */
+	renderFooter?: React.ReactElement;
```

Comment:
I think it makes more sense to accept function when we need to pass some extra information so it can be rendered, I do not see any other benefit besides that but I can be totally mistaken.

Reply:
In [your other comment](https://github.com/liferay/clay/pull/1842#discussion_r278552133) you said:

> The dev might have more control over Footer (but not so much...), previously the footer was defined by an API. That's why more freedom. I will improve this.

Which makes it sound like you want more flexibility. There's obviously a trade-off here between control (doing only what is narrowly and explicitly defined in Lexicon) and flexibility (being able to go beyond what's explicitly specified). We should just try to be clear about which end of that spectrum we want to stand at.

Reply:
I do not quite understand what my comment has to do with this thread 😅 (maybe I'm understanding this wrong) but yes you're right about that, we should be careful with the way we want to position, when I say this is the in case you had no control before, now we are with a little freedom for you to set your buttons components or turn into something else ... without needing the dependency of a defined API to do such but still continue on the surveillance of the specification (which is the result of the rules).

Reply:
Well it's about where we fall on the spectrum:

before: no control -> in this PR: some more control -> allowing a function: even more control

Just wanting us to be intentional about where we land. My sense is that in this project, because it self-identifies as an implementation of Lexicon, that we have generally leant towards the left side. My habit has long been to lean in the other direction (not by adding a million options, but by doing things like allowing functions in certain places because that effectively constitutes an "escape hatch"). The problem with allowing more control, of course, is that you can wind up in a situation [like this](https://xkcd.com/1172/) where you lose the freedom to change things that people come to depend on.

Reply:
Hmm yeah, makes sense. We were really leaning towards the Lexicon, I think we are taking short steps to give more flexibility and doors of escape like this, I think it is good to go slowly as we go slowly, and we can reach a middle ground without extrapolating.